dev/com.ibm.ws.webcontainer.servlet.3.1/src/com/ibm/ws/webcontainer31/util/UpgradeInputByteBufferUtil.java

```diff
@@ -164,51 +161,75 @@ private boolean syncRead(int amountToRead) throws IOException{
      */
     private boolean immediateRead(int amountToRead){
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()){
-            Tr.debug(tc, "immediateRead, Executing a read");
+            Tr.debug(tc, "immediateRead ENTER , buffer size [" + amountToRead + "]");
         }
         
         if(amountToRead > 1){
-          //Allocate a new temp buffer, then set the position to 0 and limit to the amount we want to read
-            //Copy in the current this.buffer as it should only have one byte in it
-            WsByteBuffer tempBuffer = allocateBuffer(amountToRead);
-            tempBuffer.position(0);
-            tempBuffer.limit(amountToRead);
-            tempBuffer.put(_buffer);
-            tempBuffer.position(1);
-            _buffer.release();
-            _buffer = tempBuffer;
-            tempBuffer = null;
-            
-            _tcpContext.getReadInterface().setBuffer(_buffer);
-            
-            long bytesRead = 0;
-            
-            try{
-                bytesRead = _tcpContext.getReadInterface().read(0, WCCustomProperties31.UPGRADE_READ_TIMEOUT);
-            } catch (IOException readException){
-                //If we encounter an exception here we need to return the 1 byte that we already have.
-                //Returned true immediately and the next read will catch the exception and propagate it properly
+            /*
+             * On a slow system/startup, data comes from the wire is faster than the channel can read/parse.  
```

Comment:
"On a slow system/startup, data comes from the wire faster than the channel can read/parse."